### DISCUSSION TITLE
Component "Upload": Allow Preview, Crop and Rotate

### DISCUSSION THREAD

**Discussion by turolopezsanabria** [SUI-Components/sui-components#1501] [Ideas and new features]
**Description**
There's a need to create a new variation of the component Upload, allowing basic edition crop and rotate.

> THE question is: Is it a complete new component, or it has to be an iteration of the current component "upload"?

**Discussion and actions**

-  **Is it Upload or Photo-Uploader to use this new functionality?**
[Upload](https://sui-components.vercel.app/workbench/atom/upload)

-  **Should it be a different component or a new variation of the existing one?**
Variation of the existing one

- **Is part of the new functionality to edit OLD images, already uploaded before?**
Not contemplated, at least initially

- **Is it ratio, or exact size / dimension?**
Ratio proportion

- **Will it crop and edit the image before loading and upload only the final result, or we keep the original img?**
We upload the result, we won't save the original

- **What do we do with formats that don't allow cropping? (pdf for instance)**
We ask if the component will allow cropping or not (is this a prop?).
If true, the component will allow only the valid format of images.

- **What do we do with the preview?**
The process must have more than 1 step: 1 Stage (image is selected and ready to manipulate) > 2 Edit and preview > 3 Upload to final destination

- **Can't we just add a background to the container?**
We are doing that but the fill is white. We could improve it but it wouldn't fix our problem for all cases.

**Specs**
tbd

**More info:**
Check the kickoff discussion during our [weekly sync](https://www.twitch.tv/videos/1021007474) on Thursday 13.

**Comment by nuria-isart** [comment]
After discussing it, we decide that the best guess will be create a new component "Image Editor".

**Comment by adriadoz** [comment]
Are we talking about `atom/Upload` or `molecule/PhotoUploader`?
The first one is for documents, if we are talking about croping and image it is the second one?

**Comment by turolopezsanabria** [comment]
Yes!
We have it as a new component "Image Editor", you can see it here:
https://sui-components.vercel.app/workbench/molecule/imageEditor/demo
